surge/api_resource.py

```diff
@@ -29,7 +27,7 @@ def _base_request(cls, method, api_endpoint, params=None):
             raise SurgeMissingAPIKeyError
 
         try:
-            url = f"{BASE_URL}/{api_endpoint}"
+            url = f"{surge.base_url}/{api_endpoint}"
```

Comment:
It took me a bit to figure it out but it will actually -- not sure how to link to the other change but I'll comment there. But basically `surge/__init__.py` becomes `surge` here and within that it has:

```python
base_url = os.environ.get("SURGE_BASE_URL", "https://app.surgehq.ai/api")
```

Which falls back to `https://app.surgehq.ai/api` unless environment has `SURGE_BASE_URL` set. I tested this manually a bit to be sure.